api/src/main/java/org/openmrs/api/impl/LocationServiceImpl.java

```diff
@@ -118,21 +118,13 @@ public Location getDefaultLocation() throws APIException {
 		
 		if (StringUtils.hasText(locationGP)) {
 			location = Context.getLocationService().getLocation(locationGP);
+		} else if(location == null){
+				location = getDefaultLocation(null, locationGP);	
+			}
+			return location;
```

Comment:
This is a pretty drastic change in the logic here and I'm not sure this is what we want to do. 

So before your change, this would first check if `locationGP` has text and if it does, try to load that location. Then if `locationGP` didn't have text or the the location specified by `locationGP` couldn't be found, it would try to get the default location by calling the private `getDefaultLocation()` function. If none of those methods returned a location, it would try to get Location 1 (whatever that is) and return it.

With this change, this will check if `locationGP` has text and if it does, it attempts to return that location. If `locationGP` does not have text (the check on `location == null` is actually redundant because at this point, we know `location` is always `null`), it will call the private `getDefaultLocation()` function. The function will then return the result of either `getLocation(locationGP)` (if `locationGP` has text) or the result of `getDefaultLocation(null, locationGP)` (if `locationGP` does not have text.

In other words, I could simplify the logic in your change to something like this:

```java
return StringUtils.hasText(locationGP) ? Context.getLocationService().getLocation(locationGP) : getDefaultLocation(null, "");
```

I don't think there's much to gain from trying to simplify this method much more than it already is... At a certain point, methods get too "simple" and the logic becomes hard to follow.

Reply:
Haaa thanks for the logical testing and logical clarification,i will try to do some changes as you suggested

Reply:
How does it look now,Am not so conviced of how this methods were called from the start btw